server/server.go

```diff
@@ -105,6 +108,27 @@ func handler(s ssh.Session) {
 	verbose("handler exits")
 }
 
+func Advertise(port, txt string) (*mdns.Server, error) {
+	v("starting mdns server")
+	// Setup our service export
+	host, err := os.Hostname()
+	if err != nil {
+		return nil, fmt.Errorf("Could not determine hostname: %w", err)
+	}
+	p, err := strconv.ParseUint(port, 0, 0)
+	if err != nil {
+		return nil, fmt.Errorf("Could not parse port: %s, %w", port, err)
+	}
+	service, err := mdns.NewMDNSService(host, "_ncpu._tcp", "", "", int(p), nil, strings.Split(txt, ","))
+	if err != nil {
+		return nil, fmt.Errorf("Could not instantiate mDNS server on port %d: %w", p, err)
+	}
+
+	server, err := mdns.NewServer(&mdns.Config{Zone: service})
```

Comment:
You can do `return mdns.NewServer(...)` here instead of `server, err := ...; return server, err`

Reply:
will simplify.